contents/docs/self-host/runbook/kafka/log-retention.mdx

```diff
@@ -0,0 +1,14 @@
+---
+title: Log retention
+sidebar: Docs
+showTitle: true
+---
+
+When Kafka's disk gets full, the service can get stuck, leading us to drop all incoming events. To mitigate the issue, we can edit Kafka's log retention policies to free up some space. There are two configs we can set (both set minimum values and data can't be deleted beforehand):
+
+1. time - [kafka docs](https://kafka.apache.org/documentation/#topicconfigs_local.retention.ms)
+2. bytes - [kafka docs](https://kafka.apache.org/documentation/#topicconfigs_local.retention.bytes)
+
+Note that the retention check loop by default is ran every 5min [retention check interval](https://kafka.apache.org/documentation/#brokerconfigs_log.retention.check.interval.ms), we can change it to be more frequent, but probably don't need to.
+
+We want to minimize the probability of kafka disk getting full and losing events and maximize disk usage to have as long retention as possible to have data to recover from in case something broke about ingestion. Therefore we suggest setting time to be relatively low (2h or 24h) and bytes to be about 90% of the volume size.
```

Comment:
Should we add a section on how to change those values? Afaik updating the config will only change the default value used by the Kafka broker when creating a new topic. To modify an already existing topic we need to make a KV update in zookeeper.

Reply:
Looks like it did get updated when I changed the config & ran a helm install, verified in the Kafka pod:
```
/opt/bitnami/kafka/bin/kafka-topics.sh --bootstrap-server localhost:9092 --describe
Topic: clickhouse_events_proto	PartitionCount: 1	ReplicationFactor: 1	Configs: flush.ms=1000,segment.bytes=1073741824,flush.messages=10000,max.message.bytes=1000012,retention.bytes=4500000000
```